### ISSUE TITLE
Feature request: Ability to publish 'distributions' as a package

### ISSUE THREAD

**Issue by samjarrett** [composer/composer#5892]
Lots of modern frameworks ([symfony](https://github.com/symfony/symfony-standard), drupal[[1](https://packagist.org/packages/drupal/drupal), [2](https://github.com/drupal-composer/drupal-project)], [wordpress' bedrock](https://github.com/roots/bedrock)) publish a 'distribution', repository that includes a `composer.json` and lock file for dependencies.   

Typically a user's workflow to create a project with one of these frameworks starts with `composer create-project vendor/distribution-package`, which creates an entire clone of the repo which then triggers a dependency install of those listed in composer.json.

When you upgrade versions of the frameworks though, you often have to go back and re-look at the distribution repo and see what dependencies and version restrictions to add to your project.

It would be helpful if Composer introduced a way to blindly pull in a distribution's requirements (either as a `require` or as something else) and possibly lock to their exact versions in the published `composer.lock` file to make it easier for users to update.

Currently it is possible to add a `require` entry for frameworks, e.g. `"symfony/framework-standard-edition": "~3.1.7"`, but this has two main drawbacks:

- the distribution's directory structure is replicated inside the `vendor/path/to/dist` directory
- the distribution's `composer.lock` file is ignored

While a workaround could be achieved by framework maintainers publishing an empty package for the dependencies of their distribution and locking their dependencies to specific versions, it feels inelegant and that a composer-native/approved solution would be better.

**Comment by alcohol** [comment]
TL/DR: Pass `--keep-vcs` to `create-project`.

Additionally, we will not support this. Composer deals with dependencies, not with your root project. You control your root project, and you maintain it. You might have build it on top of some "distribution" of your favorite framework, but you add in your own code and what not, and there is no way for us to manage all of this cleanly. Especially since each framework does things their own way. The closest you can get to a solution is using `--keep-vcs`, which will keep the `.git` directory of the distribution you picked, allowing you to simply `git pull` new changes.

**Comment by stof** [comment]
the fact is that it depends of what you try to do. ``symfony/framework-standard-edition`` is precisely the bad example for your feature request, because this package was never meant to be required in projects. It was always meant as a bootstrap for new projects. Once it has been used, people using Symfony are free to make any change they want.
